src/sentry/contexts.cpp

```diff
@@ -106,6 +107,24 @@ Dictionary make_device_context(const Ref<RuntimeConfig> &p_runtime_config) {
 	return device_context;
 }
 
+Dictionary make_device_context_update() {
+	Dictionary device_context = Dictionary();
+	int primary_screen = DisplayServer::get_singleton()->get_primary_screen();
+	device_context["orientation"] = _screen_orientation_as_string(primary_screen);
+
+	const Dictionary &meminfo = OS::get_singleton()->get_memory_info();
+	// Note: Using double since int32 can't handle size in bytes.
+	device_context["free_memory"] = double(meminfo["free"]);
+	device_context["usable_memory"] = double(meminfo["available"]);
+
+	auto dir = DirAccess::open("user://");
```

Comment:
Do we need permissions for that?

Reply:
It's guaranteed to be writable, judging by the docs: https://docs.godotengine.org/en/stable/tutorials/io/data_paths.html#accessing-persistent-user-data-user

Also, some interesting bits:
> On mobile platforms, this path is unique to the project and is not accessible by other applications for security reasons.

> On HTML5 exports, user:// will refer to a virtual filesystem stored on the device via IndexedDB. (Interaction with the main filesystem can still be performed through the [JavaScriptBridge](https://docs.godotengine.org/en/stable/classes/class_javascriptbridge.html#class-javascriptbridge) singleton.)